html-aam/index.html

```diff
@@ -14589,6 +14642,58 @@ <h4 id="att-step">`step`</h4>
             </tr>
           </tbody>
         </table>
+        <h4 id="att-switch">`switch`</h4>
+        <table aria-labelledby="att-switch">
+          <tbody>
+            <tr>
+              <th>HTML Specification</th>
+              <td>`switch`</td>
+            </tr>
+            <tr>
+              <th>Element(s)</th>
+              <td><a data-cite="html/input.html#attr-input-checked">`input`</a> `type=checkbox`</td>
+            </tr>
+            <tr>
+              <th>[[WAI-ARIA-1.2]]</th>
+              <td>
+                <div class="general">Not mapped</div>
+              </td>
+            </tr>
+            <tr>
+              <th>
+                <a href="https://msdn.microsoft.com/en-us/library/dd373608%28v=VS.85%29.aspx">MSAA</a> + <a href="http://accessibility.linuxfoundation.org/a11yspecs/ia2/docs/html/">IAccessible2</a>
+              </th>
+              <td>
+                <div class="general">Not mapped</div>
+              </td>
+            </tr>
+            <tr>
+              <th><a href="https://msdn.microsoft.com/en-us/library/ms726297%28v=VS.85%29.aspx">UIA</a></th>
+              <td>
+                <div class="general">Not mapped</div>
+              </td>
+            </tr>
+            <tr>
+              <th><a href="https://gnome.pages.gitlab.gnome.org/atk/">ATK</a></th>
+              <td>
+                <div class="general">Not mapped</div>
+              </td>
+            </tr>
+            <tr>
+              <th><a href="https://developer.apple.com/reference/appkit/nsaccessibility">AX</a></th>
+              <td>
+                <div class="general">Not mapped</div>
```

Comment:
cross link the WebKit AX PR for the relevant change. @annevk worked on it IIRC.

Reply:
Let's see, there's AAM tests here:

- https://github.com/web-platform-tests/wpt/blob/master/html-aam/roles-dynamic-switch.tentative.window.js
- https://github.com/web-platform-tests/wpt/blob/master/html-aam/roles-generic.tentative.html

We made several changes to WebKit:

- Made it map to `role=switch`: https://github.com/WebKit/WebKit/pull/18897.
- Made `role=switch` (which `<input type=checkbox switch>` builds on) support the notion of being required, similar to `role=checkbox`. That happened in https://github.com/WebKit/WebKit/pull/18938.
- Calculating rectangles correctly in the a11y code: https://github.com/WebKit/WebKit/pull/19048.

We also made a change to VoiceOver (change itself not public): https://bugs.webkit.org/show_bug.cgi?id=268120.

Hope that helps!

Reply:
@cookiecrook this PR is not missing the platform mappings?   They all say to use the wai-aria mapping for the switch role (though that was incorrectly linking to checkbox and not switch role, just fixed that) in the input (type attribute in the Checkbox state) with the switch attribute mapping table.

the switch attribute itself just points up to that role table (this spec does similar for the `list` attribute on `input` elements where the existence of the attribute changes the input's role to `combobox`).  "Not mapped" is probably not the right text for the switch attribute table though, and instead each row should just indicate the attribute changes the role of the input.  i'll make that change.